src/librustdoc/formats/mod.rs

```diff
@@ -0,0 +1,107 @@
+use std::cell::RefCell;
+use std::rc::Rc;
+
+use rustc_span::edition::Edition;
+
+use crate::clean;
+use crate::config::{RenderInfo, RenderOptions};
+use crate::error::Error;
+
+pub trait FormatRenderer: Clone {
+    type Output: FormatRenderer;
```

Comment:
Just curious, would this ever be anything different than Self? Does it need to be an associated type?

Reply:
Nope, good catch. That was an artifact of a previous version.